tests/integration/test_parallel_resource_executor_processor.py

```
@@ -0,0 +1,740 @@
+        from file_organizer.parallel.config import ParallelConfig
+        from file_organizer.parallel.processor import ParallelProcessor
+
+        files = [tmp_path / f"e{i}.txt" for i in range(2)]
+        for f in files:
+            f.write_text("x", encoding="utf-8")
+
+        cfg = ParallelConfig(max_workers=2, retry_count=0, timeout_per_file=30.0)
+        proc = ParallelProcessor(cfg)
+
+        with ThreadPoolExecutor(max_workers=2) as executor:
+            results = list(proc.process_batch_iter(files, lambda p: "ok", executor=executor))
```

Ruff (0.15.9)

[warning] Unused lambda argument: `p`

(ARG005)